sdk-core/src/main/java/dev/restate/sdk/core/InvocationStateMachine.java

```diff
@@ -236,12 +250,15 @@ void fail(Throwable cause) {
       if (this.outputSubscriber != null) {
         // Publish ErrorMessage to output subscriber before closing.
         if (cause instanceof ProtocolException) {
-          this.outputSubscriber.onNext(((ProtocolException) cause).toErrorMessage());
+          this.outputSubscriber.onNext(((ProtocolException) cause).toEndMessage());
         } else if (cause != null) {
           this.outputSubscriber.onNext(
-              Protocol.ErrorMessage.newBuilder()
-                  .setCode(Code.UNKNOWN_VALUE)
-                  .setMessage(cause.toString())
+              Protocol.EndMessage.newBuilder()
+                  .setFailure(
+                      Protocol.DetailedFailure.newBuilder()
+                          .setCode(Code.UNKNOWN_VALUE)
+                          .setMessage(cause.toString())
+                          .build())
```

Comment:
I thought that we would keep the `ErrorMessage`?

Reply:
Sorry, I updated the PR now.